src/odemis/driver/focustracker.py

```diff
@@ -39,171 +44,180 @@ def __init__(self, name, role, channel, node_idx, **kwargs):
         model.HwComponent.__init__(self, name, role, **kwargs)
         # Connect to the CANbus and the CANopen network.
         self.network = canopen.Network()
-        self.network.connect(bustype='socketcan', channel=channel)
+        bustype = 'socketcan' if channel != 'fake' else 'virtual'
+        try:
+            self.network.connect(bustype=bustype, channel=channel)
+        except IOError as exp:
+            if exp.errno == 19:
+                raise HwError("Focus Tracker is not connected.")
+            else:
+                raise
         self.network.check()
-        self.node = canopen.RemoteNode(node_idx, os.path.dirname(odemis.__file__) + "/driver/FocusTracker.eds")
+        object_dict = pkg_resources.resource_filename("odemis.driver", "FocusTracker.eds")
+        if channel == 'fake':
+            self.node = FakeRemoteNode(node_idx, object_dict)
+        else:
+            self.node = canopen.RemoteNode(node_idx, object_dict)
         self.network.add_node(self.node)
+        self._swVersion = "python-canopen v%s , python-can v%s" % (canopen.__version__, can.__version__)
+        rev_num = hex(self.node.sdo["Identity Object"]["Revision number"].raw).split('x')[-1]
+        major = int(rev_num[:-4])
+        minor = int(rev_num[-4:])
+        self._hwVersion = "{}.{}".format(major, minor)
```

Comment:
That's not going to work well if there is any number above 9 (as 10 is expressed as "a"). You could use `int(...,16)`. However, you can also directly convert to the right number with such code:
```
rev_num = self.node.sdo["Identity Object"]["Revision number"].raw
minor = rev_num & 0xffff
major = (rev_num >> 16) & 0xffff
```